sdks/python/apache_beam/runners/worker/sdk_worker_test.py

```diff
@@ -62,29 +62,65 @@ def Control(self, response_iterator, context):
 
 class SdkWorkerTest(unittest.TestCase):
 
-  def test_fn_registration(self):
-    process_bundle_descriptors = [
+  def _get_process_bundles(self, prefix, size):
+    return [
         beam_fn_api_pb2.ProcessBundleDescriptor(
-            id=str(100+ix),
+            id=str(str(prefix) + "-" + str(ix)),
             transforms={
-                str(ix): beam_runner_api_pb2.PTransform(unique_name=str(ix))})
-        for ix in range(4)]
-
-    test_controller = BeamFnControlServicer([beam_fn_api_pb2.InstructionRequest(
-        register=beam_fn_api_pb2.RegisterRequest(
-            process_bundle_descriptor=process_bundle_descriptors))])
+                str(ix): beam_runner_api_pb2.PTransform(unique_name=str(ix))
+            }) for ix in range(size)
+    ]
+
+  def _check_fn_registration_multi_request(self,
+                                           request_count,
+                                           process_bundles_per_request,
+                                           pipeline_options=None):
+    requests = []
+    process_bundle_descriptors = []
+
+    for i in range(request_count):
+      pbd = self._get_process_bundles(i, process_bundles_per_request)
+      process_bundle_descriptors.extend(pbd)
+      requests.append(
+          beam_fn_api_pb2.InstructionRequest(
+              instruction_id=str(i),
+              register=beam_fn_api_pb2.RegisterRequest(
+                  process_bundle_descriptor=process_bundle_descriptors)))
+
+    test_controller = BeamFnControlServicer(requests)
 
     server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
     beam_fn_api_pb2_grpc.add_BeamFnControlServicer_to_server(
         test_controller, server)
     test_port = server.add_insecure_port("[::]:0")
     server.start()
 
-    harness = sdk_worker.SdkHarness("localhost:%s" % test_port)
+    harness = sdk_worker.SdkHarness(
+        "localhost:%s" % test_port, pipeline_options=pipeline_options)
     harness.run()
-    self.assertEqual(
-        harness.worker.fns,
-        {item.id: item for item in process_bundle_descriptors})
+
+    for worker in harness.workers.queue:
+      self.assertEqual(worker.fns,
+                       {item.id: item
+                        for item in process_bundle_descriptors})
+    return harness
+
+  def test_fn_registration(self):
```

Comment:
Can we unify all these unit tests into a one, which calls _check_fn_registration_multi_request for an array of arguments.

Reply:
Sure!